dashboard/src/app/dashboard/[dashboardId]/(dashboard)/WeeklyHeatmapSection.tsx

```diff
@@ -0,0 +1,165 @@
+'use client';
+
+import { Fragment, use, useMemo, useState, type CSSProperties, useCallback } from 'react';
+import { fetchWeeklyHeatmapAllAction } from '@/app/actions/weeklyHeatmap';
+import type { HeatmapMetric } from '@/entities/weeklyHeatmap';
+import { Select, SelectContent, SelectItem, SelectTrigger, SelectValue } from '@/components/ui/select';
+import { cn } from '@/lib/utils';
+import { type WeeklyHeatmapMatrix, type WeeklyHeatmapPrepared } from '@/presenters/toWeeklyHeatmapMatrix';
+import { Tooltip, TooltipContent, TooltipTrigger } from '@/components/ui/tooltip';
+import { formatDuration } from '@/utils/dateFormatters';
+
+type WeeklyHeatmapSectionProps = {
+  weeklyHeatmapAllPromise: ReturnType<typeof fetchWeeklyHeatmapAllAction>;
+};
+
+const metricOptions = [
+  { value: 'pageviews', labelKey: 'pageviews' },
+  { value: 'unique_visitors', labelKey: 'uniqueVisitors' },
+  { value: 'sessions', labelKey: 'sessions' },
+  { value: 'bounce_rate', labelKey: 'bounceRate' },
+  { value: 'pages_per_session', labelKey: 'pagesPerSession' },
+  { value: 'session_duration', labelKey: 'sessionDuration' },
+] as const;
+
+export default function WeeklyHeatmapSection(props: WeeklyHeatmapSectionProps) {
+  const allData = use(props.weeklyHeatmapAllPromise);
+  const [selectedMetric, setSelectedMetric] = useState<HeatmapMetric>('unique_visitors');
+
+  const current: WeeklyHeatmapPrepared | undefined = useMemo(() => {
+    const pair = allData.find(([metric]) => metric === selectedMetric);
+    return pair ? pair[1] : undefined;
+  }, [allData, selectedMetric]);
+
+  const selectedMetricLabel = {
+    pageviews: 'Pageviews',
+    unique_visitors: 'Unique Visitors',
+    sessions: 'Sessions',
+    bounce_rate: 'Bounce Rate',
+    pages_per_session: 'Pages / Session',
+    session_duration: 'Session Duration',
+  }[selectedMetric];
+
+  const onMetricChange = (next: string) => {
+    setSelectedMetric(next as HeatmapMetric);
+  };
+
+  return (
+    <div className='bg-card border-border rounded-lg border p-6 shadow'>
+      <div className='mb-8 flex items-center justify-between'>
+        <div>
+          <h2 className='text-foreground mb-1 text-lg font-bold'>Weekly Trends</h2>
+        </div>
+        <div className='w-36'>
+          <Select value={selectedMetric} onValueChange={onMetricChange}>
+            <SelectTrigger className='w-full'>
+              <SelectValue />
+            </SelectTrigger>
+            <SelectContent>
+              {metricOptions.map((opt) => (
+                <SelectItem key={opt.value} value={opt.value}>
+                  {
+                    {
+                      pageviews: 'Pageviews',
+                      unique_visitors: 'Unique Visitors',
+                      sessions: 'Sessions',
+                      bounce_rate: 'Bounce Rate',
+                      pages_per_session: 'Pages / Session',
+                      session_duration: 'Session Duration',
+                    }[opt.value]
```

Comment:
Suggestion: Create the "base" object and use it for both the `selectedMetricLabel ` and for this 😄